src/main/scala/com/horizon/exchangeapi/BusinessRoutes.scala

```diff
@@ -564,11 +564,25 @@ trait BusinessRoutes extends JacksonSupport with AuthenticationSupport {
                 Note about Slick usage: joinLeft returns node rows even if they don't have any agreements (which is why the agreement cols are Option() )
               */
               val oldestTime = if (reqBody.changedSince > 0) ApiTime.thenUTC(reqBody.changedSince) else ApiTime.beginningUTC
-              val nodeQuery =
-                for {
+              logger.debug("ident.isAdmin: " + ident.isAdmin)
+              logger.debug("ident.role.equals(AuthRoles.Agbot): " + ident.role.equals(AuthRoles.Agbot))
+              logger.debug("ident.role: " + ident.role)
+              logger.debug("(ident.isAdmin || ident.role.equals(AuthRoles.Agbot)): " + (ident.isAdmin || ident.role.equals(AuthRoles.Agbot)))
+              logger.debug("ident.getIdentity: " + ident.getIdentity)
+              if (ident.isAdmin || ident.role.equals(AuthRoles.Agbot)) {
```

Comment:
I think i agree with sadiyah that it will be much less confusing to the non-admin user if we prevent him from running this route vs. let him run it but return less nodes.

So i think it best to remove your changes in this route, because that way `exchAuth(TNode(OrgAndId(orgid,"#").toString), Access.READ)` will reject a non-admin user.

Reply:
Business policy search changes have been reverted.